### DISCUSSION TITLE
Issues with reset sudo timestamp added in #17694

### DISCUSSION THREAD

**Discussion by viking1304** [Homebrew/discussions#5528] [Everyday usage]
### Output of `brew config`

```shell
HOMEBREW_VERSION: 4.3.9-462-g807a934
ORIGIN: https://github.com/Homebrew/brew
HEAD: 807a9345744e77493fe2a433d16b468d255eb10b
Last commit: 12 hours ago
Core tap JSON: 28 Jul 13:25 UTC
Core cask tap JSON: 28 Jul 13:25 UTC
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: ["--no-quarantine"]
HOMEBREW_EDITOR: code
HOMEBREW_GITHUB_API_TOKEN: set
HOMEBREW_MAKE_JOBS: 12
HOMEBREW_SORBET_RUNTIME: set
Homebrew Ruby: 3.3.4 => /opt/homebrew/Library/Homebrew/vendor/portable-ruby/3.3.4/bin/ruby
CPU: dodeca-core 64-bit arm_lobos
Clang: 15.0.0 build 1500
Git: 2.45.2 => /opt/homebrew/bin/git
Curl: 8.6.0 => /usr/bin/curl
macOS: 14.5-arm64
CLT: 15.3.0.0.1.1708646388
Xcode: 15.4
Rosetta 2: false
```

### Output of `brew doctor`

```shell
Your system is ready to brew.
```

### Description of issue

I was debugging strange behavior of one of my scripts that relies on Homebrew, and then I figured out that the source of my problem was this commit https://github.com/Homebrew/brew/pull/17694

I understand why you did it and agree that should be the default behavior. Still, I would appreciate it if you could consider adding an option to restore the old behavior for script use.

In my scripts, I like to ask for the password at the beginning and then periodically update the timestamp in a background job. This way, the user can just enter the password and forget about the script instead of waiting to be asked for a password somewhere in the middle of the execution. I will remove this for now, but that is not a real issue.

The bigger issue is that using a simple command like brew list will reset the sudo timestamp. Even if I install a formula that does not require sudo, the sudo timestamp will also be reset.

Am I missing something obvious here, or was this really intended?

**Comment by SMillerDev** [comment]
Any formula build can run any command. There is no way to know if something will run sudo or not. That's why it's needed before anything else is done.

**Reply by viking1304** [reply]
OK, that makes sense.

As I wrote, I am not against this change. I am just trying to figure out a way to keep my scripts user-friendly, especially the one that installs almost 80 apps using brew bundle and restores most of the settings. This can take some time, and because some casks require sudo and some default macOS settings cannot be set without sudo, I like to just enter the password at the beginning and let it work.

Based on what you wrote, what I am asking for should not be something that can be called from a formula or cask, so I am not sure what the best approach in this case would be. Any option or command could still be invoked by any formula or cask, which is exactly the thing you are trying to prevent.

**Comment by gibfahn** [comment]
Cross-linking to the workarounds posted in other issues:

https://github.com/Homebrew/brew/issues/17912#issuecomment-2258965623
https://github.com/Homebrew/brew/issues/17915#issuecomment-2283157831

Sounds like using `script` is the way to go.
